### PR TITLE
meta.versionId Incremented After Update Operation With No Changes (DSTU 3)

### PR DESCRIPTION
**Pull request by nathandoef**

### CODE REVIEW

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java [hunk 1]
```diff
@@ -705,7 +705,10 @@ private byte[] getResourceBinary(ResourceEncodingEnum encoding, String encodedRe
 						}
 					}
 				}
-
+				boolean allExtensionsRemoved = extensions.isEmpty();
+				if(allExtensionsRemoved){
+					hasExtensions = false;
+				}
```
Comment (tadgh) [inline hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java:711]: Question: Why does this solve the problem?

**Review by tadgh — APPROVED** [review]

File: hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java (hunk 1 above)
Comment (nathandoef) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java:711]: Further down (line 717) there is a check for `hasExtensions`. If there are no extensions the `meta` field is removed completely which should occur for DSTU3.

Basically if that `meta` field is not removed it is registering as a change which is causing the version to be incremented.

Comment (jamesagnew) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java:711]: Whoa! This is kinda hilarious. Great catch.

Comment (nathandoef) [reply hapi-fhir-jpaserver-base/src/main/java/ca/uhn/fhir/jpa/dao/BaseHapiFhirDao.java:711]: Thanks James

**Pull request merged**